rs/ic_os/manifest_tool/src/components_parser.rs

```diff
@@ -0,0 +1,127 @@
+use anyhow::{Context, Result};
+use regex::Regex;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+pub const COMPONENTS_PATH: &str = "ic-os/components";
+
+#[derive(Debug)]
+pub struct IcosManifest {
+    pub setupos: Manifest,
+    pub hostos: Manifest,
+    pub guestos: Manifest,
+    pub boundary_guestos: Manifest,
+}
+
+impl IcosManifest {
+    pub fn new(
+        guestos_components: Manifest,
+        hostos_components: Manifest,
+        setupos_components: Manifest,
+        boundary_guestos_components: Manifest,
+    ) -> Self {
+        IcosManifest {
+            guestos: guestos_components,
+            hostos: hostos_components,
+            setupos: setupos_components,
+            boundary_guestos: boundary_guestos_components,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Manifest {
+    pub manifest: Vec<ManifestEntry>,
+}
+
+impl Manifest {
+    pub fn new() -> Self {
+        Manifest {
+            manifest: Vec::new(),
+        }
+    }
+
+    pub fn add_entry(&mut self, component: ManifestEntry) {
+        self.manifest.push(component);
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ManifestEntry {
+    pub source: PathBuf,
+    #[allow(dead_code)]
+    pub destination: PathBuf,
+}
+
+impl ManifestEntry {
+    pub fn new(source: PathBuf, destination: PathBuf) -> Self {
+        ManifestEntry {
+            source,
+            destination,
+        }
+    }
+}
+
+fn get_manifest(manifest_path: &Path, components_path: &Path) -> Result<Manifest> {
+    let manifest_contents = fs::read_to_string(manifest_path)
+        .with_context(|| format!("Failed to read file: {:?}", manifest_path))?;
+
+    let re = Regex::new(r#"Label\("(.+?)"\): "(.+?)""#)?;
```

Comment:
I have mixed feelings about parsing bzl files from the tool. On the one hand, it's definitely the simper approach, however it feels very fragile in case we want to refactor how the components are listed (eg. Raymond mentioned that he'd like encapsulate related components, etc.). 
A sturdier (but also more complex) approach would be writing some bazel code to collect these labels and pass them as command line args.

Reply:
@frankdavid see also my thoughts here: https://github.com/dfinity/ic/pull/942#discussion_r1723113915